src/unit_tests/wazuh_modules/vulnerability_detector/test_wm_vuln_detector.c

```diff
@@ -2198,6 +2199,34 @@ void test_wm_vuldet_generate_os_and_kernel_package_release_end(void **state)
     assert_int_equal(ret, 0);
 }
 
+/* wm_vuldet_compare_vendors */
+
+void wm_vuldet_compare_vendors_external_vendor(void **state) 
+{
+    int ret;
+    char * vendor = "External";
+
+    ret = wm_vuldet_compare_vendors(vendor);
+    assert_int_equal(ret, 1);
+}
+
+void wm_vuldet_compare_vendors_official_vendor(void **state) 
+{
+    int ret;
+    int i;
+    char * official_vendors[] = {
+        "Red Hat, Inc.",
+        "CentOS",
+        "Fedora Project"
+    };
+    int array_size = sizeof(official_vendors)/sizeof(official_vendors[0]);
```

Comment:
Rebase `master` and use https://github.com/wazuh/wazuh/blob/4728cb2fe6f621d9af8a1061445b9725656529fe/src/headers/shared.h#L211